tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,259 @@
+from typing import Dict, List, Optional
+
+import pytest
+from unittest import mock
+
+from great_expectations.core.domain import Domain
+from great_expectations.core.metric_domain_types import MetricDomainTypes
+from great_expectations.rule_based_profiler.config import ParameterBuilderConfig
+from great_expectations.rule_based_profiler.helpers.util import (
+    get_parameter_value_and_validate_return_type,
+)
+from great_expectations.rule_based_profiler.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.types.attributes import Attributes
+from great_expectations.rule_based_profiler.parameter_container import (
+    ParameterContainer,
+    ParameterNode,
+    FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY,
+)
+
+
+class DummyParameterBuilder(ParameterBuilder):
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Builds ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and details.
+
+        Returns:
+            Attributes object, containing computed parameter values and parameter computation details metadata.
+        """
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
+
+@pytest.mark.unit
+@mock.patch("test_parameter_builder.DummyParameterBuilder._build_parameters")
```

Comment:
Instead of mocking this private method, can the dummy actual override the definition? If this needs to be different per test, we can parameterize with the constructor

Reply:
@cdkini We can absolutely do that -- and I had that solution, but the nI could not use the "Mock.call_count" -- can you suggest working around that?  In other words, we have to test that the implementation method was called.  Otherwise, let us leave the use of the Mock.  Please let me know what you think.  Thanks.